dataikuapi/dss/projectlibrary.py

```diff
@@ -0,0 +1,275 @@
+        self.parent._remove_child_(self)
+        destination_folder._add_child_(self)
+        self.parent = destination_folder
+
+
+class DSSLibraryFolder(DSSLibraryItem):
+    """
+    A handle to manage a library folder
+    """
+    def __init__(self, client, project_key, name, parent, children):
+        """Do not call directly, use :class:`dataikuapi.dss.projectlibrary.DSSLibrary.get_folder`"""
+        super(DSSLibraryFolder, self).__init__(client, project_key, name, parent, children)
+
+    def get_child(self, name):
+        """
+        Retrieve the sub item by its name
+
+        :param: str name: the name of the sub item
+        :returns: the sub item
+        :rtype: :class:`dataikuapi.dss.projectlibrary.DSSLibraryItem`
+        """
+        for c in self.children:
+            if c.name == name:
+                return c
+        return None
+
+    def add_file(self, file_name):
+        """
+        Create a file in the library folder
+
+        :param: str file_name: the file name
+        :returns: the new file
+        :rtype: :class:`dataikuapi.dss.projectlibrary.DSSLibraryFile`
+        """
+        if "/" in file_name:
+            raise DataikuException("File name %s contains invalid character '/'" % file_name)
+        new_path = self._clean_path_(self.get_path() + "/" + file_name)
+        self.client._perform_empty("POST", "/projects/%s/libraries/contents/%s" % (self.project_key, new_path))
+        file = DSSLibraryFile(self.client, self.project_key, file_name, self)
+        self.children.add(file)
+        return file
+
+    def add_folder(self, folder_name):
+        """
+        Create a folder in the library
+
+        :param: str folder_name: the name of the folder to add
+        :returns: the new folder
+        :rtype: :class:`dataikuapi.dss.projectlibrary.DSSLibraryFolder`
+        """
+        if "/" in folder_name:
+            raise DataikuException("Folder name %s contains invalid character '/'" % folder_name)
+        new_path = self._clean_path_(self.get_path() + "/" + folder_name)
+        self.client._perform_empty("POST", "/projects/%s/libraries/folders/%s" % (self.project_key, new_path))
+        folder = DSSLibraryFolder(self.client, self.project_key, folder_name, self, set())
+        self.children.add(folder)
+        return folder
+
+    def list(self):
+        """
+        Gets the contents of this folder sorted by name
+
+        :returns: a sorted list of items
+        :rtype: list of :class:`dataikuapi.dss.projectlibrary.DSSLibraryItem`
+        """
+        return sorted(list(self.children), key=lambda x: x.name)
+
+    def _add_child_(self, child):
+        """
+        Adds the given child from this folder
+        """
+        self.children.add(child)
+
+    def _remove_child_(self, child):
+        """
+        Removes the given child from this folder
+        """
+        self.children.remove(child)
+
+
+class DSSLibraryFile(DSSLibraryItem):
```

Comment:
Maybe we could store the content of the file in a prop, `content`
get_content()  -> do the call to get the file contents and update `content`
set_content() -> overwrite `content`
save() -> do the POST request to save the file 

WDYT?